### PR TITLE
-pthread implies USE_PTHREADS=1 and vice versa

### LINKED ISSUE #8536: make `-s USE_PTHREADS` imply `-pthread` and vice versa
When using the LLVM backend.

### PR DESCRIPTION
**Pull request by tlively**
Fixes #8536.

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -7636,6 +7636,8 @@ def setUp(self):
       # propagate LLVM machine arguments
       if arg.startswith('-m'):
         Building.COMPILER_TEST_OPTS.append(arg)
+      if arg == '-pthread':
+        Building.COMPILER_TEST_OPTS.append(arg)
```
Comment (sbc100) [inline tests/test_core.py]: Is this needed?

File: emcc.py [hunk 2]
```diff
@@ -2525,6 +2527,9 @@ def check_bad_eq(arg):
       settings_changes.append('SIMD=1')
     elif newargs[i] == '-mno-simd128':
       settings_changes.append('SIMD=0')
+    # Record USE_PTHREADS setting because it controls whether --shared-memory is passed to lld
```
Comment (sbc100) [inline emcc.py]: Probably not the only reason.  I would drop this comment and instead modify the docs for USE_PTHREADS in `src/settings.py` to mention the fact the `-pthread` does the same thing?

Comment (tlively) [reply emcc.py]: Good idea.

**Comment by tlively** [comment]
What might a test look like? This PR isn't meant to make -pthread actually work, it's just meant to make it break a little less.

**Comment by sbc100** [comment]
Does this mean we are not currently testing threads with the wasm backend?  I guess not?    If thats the case we should try to start doing that, but i won't block this change on that.

**Pull request merged**